python/rascaline/rascaline/utils/clebsch_gordan/_lambda_soap.py

```diff
@@ -0,0 +1,288 @@
+"""
+Module containing a convenience calculator that returns the output of a CG tensor
+product between :py:call:`LambdaSoap` and :py:class:`SphericalExpansionByPair`
+equivariant descriptors.
+"""
+
+from typing import List, Optional, Union
+
+import numpy as np
+
+from .. import _dispatch
+from .._backend import (
+    Labels,
+    SphericalExpansion,
+    SphericalExpansionByPair,
+    TensorBlock,
+    TensorMap,
+    TorchModule,
+    TorchScriptClass,
+    operations,
+    torch_jit_export,
+    torch_jit_is_scripting,
+)
+from . import _coefficients, _utils
+from ._correlate_density import DensityCorrelations
+from ._correlate_tensors import CorrelateTensorWithDensity
+
+
+try:
+    import torch
+
+    HAS_TORCH = True
+except ImportError:
+    HAS_TORCH = False
+
+# ======================================================================
+# ===== Public API functions
+# ======================================================================
+
+
+class LambdaSoap(TorchModule):
+    """
+    Computes a Lambda-SOAP equivariant descriptor.
+    """
+
+    def __init__(
+        self,
+        spherical_expansion_hypers: dict,
+        atom_types: Optional[List[int]] = None,
+        angular_cutoff: Optional[int] = None,
+        selected_keys: Optional[Labels] = None,
```

Comment:
this should be part of `compute/forward`: different calls to the same calculator might request different keys